policies/queries/sns/topics_unused.sql

```diff
@@ -0,0 +1,14 @@
+insert into aws_policy_results
+with subscription as (select distinct topic_arn from aws_sns_subscriptions)
+select :'execution_time'                     as execution_time,
+       :'framework'                          as framework,
+       :'check_id'                           as check_id,
+       'Unused SNS topics should be removed' as title,
+       topic.account_id,
+       topic.arn                             as resource_id,
+       case
+           when subscription.topic_arn is null then 'fail'
+           else 'pass'
+           end                               as status
```

Comment:
I don't think you can have a subscription without a topic.

When you create a subscription, the Topic ARN is a required field and when you delete a topic all subscriptions get deleted

Reply:
here we check for topics that have no subscriptions, not the other ay around